### DISCUSSION TITLE
A better way to query branch metadata?

### DISCUSSION THREAD

**Discussion by alexpghayes** [ropensci/targets#420] [Help]
When I use dynamic within static branching, I find myself rather frequently using regex on target names to recover metadata that I want to use. I use the branching to iterate over estimators/models/data sources and then I produce a bunch of diagnostics for each estimator/model/data source (static branching), and for each of these I have a bunch of hyperparameters that I iterate over with dynamic branching. It is easy to get the hyperparameter information / dynamic branch provenance, I just do something like

``` r
tar_target(
      hub_tables,
      extract_hubs(vsp_analysis, node_data, hyperparameters),
      pattern = map(vsp_analysis, hyperparameters),
      format = "file"
)
```

To get the static branch meta data I do something like the following within `extract_hubs()`:

``` r
  metadata_list <- unglue(tar_name(), "{junk}_{estimator}_{data}_{hash=[a-z0-9]*}")
```

Is there some approach here that doesn't involve parsing target names like this?

**Comment by wlandau** [comment]
I recommend keeping track of the metadata inside the targets you map over, e.g. `vsp_analysis` or `hyperparameters`, so you do not need to interpret target names.
